File: email-creator/src/main/java/dev/andzwp/emailcreator/config/RabbitMQConfiguration.java
package dev.andzwp.emailcreator.config;

import org.springframework.amqp.core.*;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class RabbitMQConfiguration {

    @Value("${rabbit.email.queue}")
    private String queueName;

    @Value("${rabbit.email.exchange}")
    private String topicExchangeProducerName;


    @Bean
    public Queue queue() {
        return new Queue(queueName, true);
    }

    @Bean
    public FanoutExchange exchange() {
        return new FanoutExchange(topicExchangeProducerName);
    }

    @Bean
    public Binding binding(Queue queue, FanoutExchange exchange) {
        return BindingBuilder.bind(queue).to(exchange);
    }
}

File: email-sender/src/main/kotlin/dev/andzwp/emailsender/config/RabbitMQConfiguration.kt
package dev.andzwp.emailsender.config

import org.springframework.amqp.core.Binding
import org.springframework.amqp.core.BindingBuilder
import org.springframework.amqp.core.Queue
import org.springframework.amqp.core.TopicExchange
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class RabbitMQConfiguration(
    @Value("\${rabbitmq.queue.name}") private val queueName: String,
    @Value("\${rabbitmq.exchange.name}") private val topicExchangeName: String,
    @Value("\${rabbitmq.routing.key}") private val routingKey: String
) {
    @Bean
    fun queue(): Queue =
        Queue(queueName)

    @Bean
    fun exchange(): TopicExchange =
        TopicExchange(topicExchangeName)

    @Bean
    fun binding(queue: Queue, exchange: TopicExchange): Binding =
        BindingBuilder.bind(queue).to(exchange).with(routingKey)
}